static/app/views/issueDetails/groupTags.tsx

```diff
@@ -1,195 +1,167 @@
-              includeAll: true,
-              query: getSampleEventQuery({
-                transaction,
-                durationBaseline: aggregateRange2,
-                addUpperBound: false,
-              }),
-              start: new Date(breakpoint * 1000).toISOString(),
-              end: new Date(this.state.now).toISOString(),
-            },
-          },
-        ],
-      ];
-    }
-
-    return [
-      [
-        'tagList',
-        `/organizations/${organization.slug}/issues/${group.id}/tags/`,
-        {
-          query: {environment: environments},
-        },
-      ],
-    ];
-  }
+type SimpleTag = {
+  key: string;
+  topValues: Array<{
+    count: number;
+    name: string;
+    value: string;
+    query?: string;
+  }>;
+  totalValues: number;
+};
 
-  componentDidUpdate(prevProps: Props) {
-    if (
-      !isEqual(prevProps.environments, this.props.environments) ||
-      !isEqual(prevProps.event, this.props.event)
-    ) {
-      this.remountComponent();
-    }
-  }
+function isTagFacetsResponse(
+  _: GroupTagsResponse | Tag[] | undefined,
+  shouldUseTagFacetsEndpoint: boolean
+): _ is Tag[] {
+  return shouldUseTagFacetsEndpoint;
+}
 
-  onRequestSuccess({stateKey, data}: {data: any; stateKey: string}): void {
-    if (stateKey === 'tagFacets') {
-      this.setState({
-        tagList: data
-          .filter(({key}) => key !== 'transaction')
-          .map(sumTagFacetsForTopValues),
-        tagFacets: undefined,
-      });
-    }
+function GroupTags({group, baseUrl, environments, event}: GroupTagsProps) {
+  const organization = useOrganization();
+  const location = useLocation();
+  const now = useMemo(() => Date.now(), []);
+
+  const {transaction, aggregateRange2, breakpoint} =
+    event?.occurrence?.evidenceData ?? {};
+
+  const shouldUseTagFacetsEndpoint =
+    organization.features.includes('performance-duration-regression-visible') &&
+    defined(event) &&
+    group.issueType === IssueType.PERFORMANCE_DURATION_REGRESSION;
+
+  const {data, isLoading, isError, refetch} = useFetchIssueTags({
+    orgSlug: organization.slug,
+    groupId: group.id,
+    environment: environments,
+    isStatisticalDetector: shouldUseTagFacetsEndpoint,
+    statisticalDetectorParameters: shouldUseTagFacetsEndpoint
+      ? {
+          transaction,
+          start: new Date(breakpoint * 1000).toISOString(),
+          end: new Date(now).toISOString(),
+          durationBaseline: aggregateRange2,
+        }
+      : undefined,
+  });
+
+  // useFetchIssueTags can return two different types of responses, depending on shouldUseTagFacetsEndpoint
+  // This line will convert the response to a common type for rendering
+  const tagList: SimpleTag[] =
+    (isTagFacetsResponse(data, shouldUseTagFacetsEndpoint)
+      ? data?.filter(({key}) => key !== 'transaction')?.map(sumTagFacetsForTopValues)
+      : data) ?? [];
+  const alphabeticalTags = (tagList ?? []).sort((a, b) => a.key.localeCompare(b.key));
```

Comment:
it looks like this is no longer required? or tagList has the wrong type
```suggestion
  const alphabeticalTags = tagList.sort((a, b) => a.key.localeCompare(b.key));
```